### PR TITLE
Enhancement/scan input autofocus

### LINKED ISSUE #318: On the note screen isbns should be accepted even with no focus on the ISBN field

### PR DESCRIPTION
**Pull request by ikusteu**
Fixes #318

### CODE REVIEW

File: pkg/ui/src/Forms/ScanInput/action.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+		// Capture the event
+		e.preventDefault();
+
+		store.update((state) => {
+			if (state.timeout) {
+				clearTimeout(state.timeout);
+			}
+
+			const buffer = state.buffer + e.key;
+			const timeout = setTimeout(store.cancel, 50);
```
[the remark this answers is not in the document]
Comment (silviot) [reply pkg/ui/src/Forms/ScanInput/action.ts:44]: @ikusteu To clarify, I don't mean it needs to be configurable via an environment variable. My point is just that it's an "important" value that might need some tweaking before finding the Goldilocks "Just right" value, and thus has dignity to have a name.

[the remark this answers is not in the document]
Comment (ikusteu) [reply pkg/ui/src/Forms/ScanInput/action.ts:44]: Yeah, I got it. Makes sense...

[the remark this answers is not in the document]
Comment (ikusteu) [reply pkg/ui/src/Forms/ScanInput/action.ts:44]: @silviot do you think I should put it next to the action (in the same folder as `constants.ts`) or should I have the action accept it as a parameter, and specify the constant, when consuming it - in the app?

[the remark this answers is not in the document]
Comment (silviot) [reply pkg/ui/src/Forms/ScanInput/action.ts:44]: I think just having it defined in the same file, at the top of the file, makes it easier to find when needed.

**Comment by ikusteu** [comment]
@silviot can I get a final amen on the PR, I've extracted the timeout into a variable.

_(I've rebased against `main`, including some flakiness fixes and other PRs merged in the past couple of hours, so the original comment is not referencing the current piece of code)_

**Review by silviot — APPROVED** [review]

**Pull request merged**